allensdk/brain_observatory/ecephys/utils.py

```diff
@@ -65,3 +66,57 @@ def clobbering_merge(to_df, from_df, **kwargs):
 
     to_df = to_df.drop(columns=list(overlapping))
     return pd.merge(to_df, from_df, **kwargs)
+
+
+def strip_substructure_acronym(
+        acronym: Optional[Union[str, list, float]]
+) -> Optional[Union[str, list]]:
+    """
+    Sanitize a structure acronym or a list of structure acronyms
+    by removing the substructure (e.g. DG-mo becomes DG).
+
+    If acronym is a list, every element in the list will be sanitized
+    and a list of unique acronyms will be returned. **Element order will
+    not be preserved**.
+
+    If acronym is None, return None. If None occurs in a list of
+    sturture acronyms, it will be omitted
+
+    Note: if acronym is NaN, it will get converted to None;
+    any other float will provoke an error.
+    """
+
+    if isinstance(acronym, numbers.Number):
```

Comment:
It looks like the outer if can be safely removed since you are just checking if it is nan here

Reply:
Actually, numpy doesn't like running `np.isnan` on non-numbers
```
>>> import numpy as np
>>> np.isnan('abcd')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: ufunc 'isnan' not supported for the input types, and the inputs could not be safely coerced to any supported types according to the casting rule ''safe''
>>> 
```